neqo-http3/src/frames/reader.rs

```diff
@@ -165,11 +165,12 @@ impl FrameReader {
         &mut self,
         stream_reader: &mut dyn StreamReader,
     ) -> Res<(Option<T>, bool)> {
+        let mut buf = [0; MAX_READ_SIZE];
```

Comment:
Given that `MAX_READ_SIZE` is `4096`, is a stack vs heap allocation the right way to go?

Reply:
Maybe. I don't have an intuition on that. In case it is not a clear win, I would prefer doing the above large refactoring instead.